src/client.ts

```diff
@@ -480,6 +483,9 @@ export class StreamChat<
       throw new Error('The "id" field on the user is missing');
     }
 
+    if (this.pendingDisconnect) await this.pendingDisconnect;
```

Comment:
I think we should include this promise in the `setUserPromise` that is used by the `queryUsers` and `queryChannels` methods to wait for pending connects